### PR TITLE
Added retry

### PR DESCRIPTION
**Pull request by lpusok**
### Context

In some cases we receive unexpected EOF errros.
Added a full download retry as a workaround until we understand this failure mode.

### CODE REVIEW

**Review by ofalvai — COMMENTED** [review]
I'm not sure it's a good idea to retry every possible cache restore error. Retrying everything 5 times can easily 5x the network transfer of the VMs if anything goes wrong and we wouldn't even notice the issue.
@viktorbenei did we run out of other ideas, is this why we want to just add a retry?

File: cache/network/download_test.go [hunk 1]
```diff
@@ -160,3 +160,76 @@ func Test_downloadFile_multipart_retrycheck(t *testing.T) {
...
+		}
+	}))
+	defer svr.Close()
+	downloadURL := svr.URL
+
+	// When
+	err := downloadFile(context.Background(), retryableHTTPClient.StandardClient(), downloadURL, tmpFile)
+
+	// Then
+	require.NoError(t, err)
```
Comment (viktorbenei) [inline cache/network/download_test.go:233]: Please also validate that retries actually did happen.

File: cache/network/download_test.go [hunk 2]
```diff
@@ -160,3 +160,76 @@ func Test_downloadFile_multipart_retrycheck(t *testing.T) {
...
+		require.NoError(t, err)
+
+		if rangeHeaderFrom == 0 && rangeHeaderTo == 0 {
+			// range request - requesting content size - return the size info
+			w.Header().Add("content-range", fmt.Sprintf("bytes 0-0/%d", len(testDummyFileContent)))
+			_, err := fmt.Fprint(w, " ")
+			require.NoError(t, err)
+		} else {
+			if atomic.LoadInt64(&numErrorsLeft) > 0 {
+				atomic.AddInt64(&numErrorsLeft, -1)
```
Comment (viktorbenei) [inline cache/network/download_test.go:209]:
```suggestion
			if numErrorsLeft.Load() > 0 {
				numErrorsLeft.Add(-1)
```

**Review by viktorbenei — CHANGES_REQUESTED** [review]
lgtm but please add a verification in the test that the retry actually happened

File: cache/network/download_test.go [hunk 3]
```diff
@@ -153,10 +155,55 @@ func Test_downloadFile_multipart_retrycheck(t *testing.T) {
...
+	var numErrorsLeft atomic.Int64
+	numErrorsLeft.Store(2)
+
+	tmpPath := t.TempDir()
+	tmpFile := filepath.Join(tmpPath, "testfile.bin")
+	testDummyFileContent := strings.Repeat("a", 10*units.MB) // 10MB
+
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Logf("[testserver] Server called. Method=%s; Header=%#v", r.Method, r.Header)
+		// range request - requesting content size - return the size info
```
Comment (viktorbenei) [inline cache/network/download_test.go:180]: left-over comment - remove

File: cache/network/download_test.go [hunk 4]
```diff
@@ -153,10 +155,55 @@ func Test_downloadFile_multipart_retrycheck(t *testing.T) {
...
+
+	// When
+	err := downloadFile(context.Background(), http.DefaultClient, downloadURL, tmpFile, logger)
+	require.NoError(t, err)
+	downloadedContents, err := os.ReadFile(tmpFile) // Read back downloaded file
+	require.NoError(t, err)
+
+	// Then
+	require.Equal(t, testDummyFileContent, string(downloadedContents), "Contents should match")
+	require.Equal(t, numErrorsLeft.Load(), int64(0), "Chunk errors should equeals content-range queries")
```
Comment (viktorbenei) [inline cache/network/download_test.go:208]: Left-over error description text. Should be something like "All error attempts depleted"

**Review by viktorbenei — CHANGES_REQUESTED** [review]
2 leftover texts, otherwise 👍 lgtm

File: cache/network/download_test.go (hunk 3 above)
Comment (lpusok) [reply cache/network/download_test.go]: removed

File: cache/network/download_test.go (hunk 4 above)
Comment (lpusok) [reply cache/network/download_test.go]: Fixed.

**Review by ofalvai — APPROVED** [review]

**Pull request merged**